create.ps1

```diff
@@ -101,278 +89,108 @@ function Get-ErrorMessage {
-            # Add correlation property to exportdata
-            $exportData | Add-Member -MemberType NoteProperty -Name $correlation.CorrelationProperty -Value $correlation.CorrelationValue -Force
-            # Add aRef properties to exportdata
-            foreach ($aRefProperty in $aRef.PSObject.Properties) {
-                $exportData | Add-Member -MemberType NoteProperty -Name $aRefProperty.Name -Value $aRefProperty.Value -Force
-            }
+    $currentAccount = $null
+    $currentAccount = Invoke-RestMethod @splatWebRequest -Verbose:$false
+  
+    if ($null -eq $currentAccount) {
+        throw = "Error querying account where [$correlationProperty)] = [$correlationValue]"
+    }
+    Write-Verbose "Correlating to account [$($currentAccount.RoepNaam) $($currentAccount.AchterNaam) ($($currentAccount.Id))]"
 
-            break
+    $outputContext.AccountReference = [PSCustomObject]@{
+        Id = $currentAccount.Id
+    }
+    # Add a message and the result of each of the validations showing what will happen during enforcement
+    if ($actionContext.DryRun -eq $true) {
+        Write-Warning "DryRun: Would correlate to account on field [$correlationProperty] with value: [$($correlationValue)]"
+    }
+    # Process
+    if (-not($actionContext.DryRun -eq $true)) {
+        
+        Write-Verbose 'Correlating user account'
+        $uri = "$($actionContext.Configuration.BaseUri)/medewerkers/$($currentAccount.Id)/$($MutationDate)"
+        $bodyobject = [PSCustomObject]@{}
+        $bodyobject | Add-Member -MemberType NoteProperty -Name EmailZakelijk -Value $actionContext.Data.EmailZakelijk -Force
+        $body = ($bodyobject | ConvertTo-Json -Depth 10) 
+          
+        $splatCorrelateParams = @{
+            Uri             = $uri
+            Headers         = $headers
+            Method          = "PUT"
+            Body            = ([System.Text.Encoding]::UTF8.GetBytes($body))
+            UseBasicParsing = $true
         }
-        #region Correlate to account
-        "Correlate" {
-            Write-Verbose "Correlating to account [$($currentAccount.RoepNaam) $($currentAccount.AchterNaam) ($($currentAccount.Id))]"
-
-            if (-not($dryRun -eq $true)) {
-                # Set aRef object for use in futher actions
-                $aRef = [PSCustomObject]@{
-                    Id = $currentAccount.Id
-                }
+        
+        $responseCorrelateAccount = Invoke-RestMethod @splatCorrelateParams -Verbose:$false 
+        
+        $auditLogMessage = "Successfully correlated account on field [$correlationProperty] with value: [$($correlationValue)]" #"$action account was successful. AccountReference is: [$($outputContext.AccountReference)"
+        $outputContext.success = $true
+        $outputContext.AccountCorrelated = $true
+        $outputContext.AuditLogs.Add([PSCustomObject]@{
+                Action  = 'CreateAccount'
```

Comment:
The action should be “CorrelateAccount” since you only correlate to the account, and no create action is performed.

Reply:
fixed